utils/zerovec/src/ule/niche.rs

```diff
@@ -8,24 +8,48 @@ use core::{marker::Copy, ops::Deref};
 
 use super::{AsULE, ULE};
 
-/// Types implementing this trait guarantee that [`NicheBytes::INVALID_BIT_PATTERN`]
+/// The types implementing this trait guarantee that [`NicheBytes::INVALID_BIT_PATTERN`]
 /// can never occur as a valid byte representation of the type.
-/// N == core::mem::sizeo_of::<Self>()
+/// The implementors guarantee that N == core::mem::sizeo_of::<Self>()
 pub trait NicheBytes<const N: usize> {
     const INVALID_BIT_PATTERN: [u8; N];
 }
 
-/// [`ULE`] type for `Option<U>` where U implements [`NicheBytes`].
-/// The invalid bit pattern is used as the niche for `Option<U>`.
-/// N == core::mem::size_of::<U>()
+/// [`ULE`] type for [`NichedOption<U,N>`] where U implements [`NicheBytes`].
+/// The invalid bit pattern is used as the niche.
+///
+/// This uses 1 byte less than [`zerovec::OptionULE<U>`] to represent [`NichedOption<U,N>`]
+/// which Derefs to [`Option<U>`].
+/// The implementors guarantee that N == core::mem::sizeo_of::<Self>()
+///
+/// Invariants:
+/// The union stores [`NichedBytes::INVALID_BIT_PATTERN`] when None.
+/// Any other bit pattern is a valid.
```

Comment:
This is an internal invariant and should not be in the doc

Reply:
Removed from the doc and left as a comment.